sharpy/linear/assembler/lineargustassembler.py

```diff
@@ -1,58 +1,551 @@
+    settings_description = {}
+
+    print_info = True  # for debugging
 
     def __init__(self):
-        self.aero = None
-        self.ss_gust = None
-        self.Kout = None
+        self.aero = None  #: aerogrid
+        self.tsaero0 = None  # timestep info at the linearisation point
+
+        self.dt = None  # type: float
+        self.remove_predictor = None  # type: bool
+        self.Kzeta = None  # type: int  # total number of lattice vertices
+        self.KKzeta = None  # type: list  # list of number of lattice vertices in each surface
+
+        self.state_to_uext = None  #type: np.array # Gust system C matrix (for unpacking into timestep_info)
+        self.uin_to_uext = None  #type: np.array # Gust system D matrix (for unpacking into timestep_info)
+        self.gust_ss = None  #type: libss.StateSpace # containing the gust state-space system
 
-    def initialise(self, aero):
+        self.settings = None
+
+        self.u_ext_direction = None  # np.ndarray Unit external velocity vector in G frame
+        self.u_inf = None  # float Free stream velocity magnitude
+
+    def initialise(self, aero, linuvlm, tsaero0, custom_settings=None):
+        """
+        Initialise gust class
+
+        Args:
+            aero (sharpy.aero.models.Aerogrid):
+            linuvlm (sharpy.linear.src.linuvlm.Dynamic) :
+            tsaero0 (sharpy.utils.datstructures.AeroTimestepInfo) : time step at reference state
+            custom_settings (dict):
+        """
         self.aero = aero
+        self.tsaero0 = tsaero0
+        self.dt = linuvlm.dt
+        self.remove_predictor = linuvlm.remove_predictor
+        self.KKzeta = linuvlm.MS.KKzeta
+        self.Kzeta = sum(self.KKzeta)
+
+        if custom_settings is not None:
+            self.settings = custom_settings
+            settings.to_custom_types(self.settings, self.settings_types, self.settings_default)
+
+        # find free stream velocity
+        u_ext = self.tsaero0.u_ext[0][:, 0, 0]  # use ref node external velocity
+        self.u_inf = np.linalg.norm(u_ext)
+        self.u_ext_direction = u_ext / self.u_inf
+
+    def get_x_max(self):
+        """
+        Returns the maximum and minimum and minimum coordinates of the UVLM lattice projected onto the free-stream
+        velocity vector
+
+        Returns:
+            tuple: minimum and maximum coordinates of lattice projected onto velocity vector
+        """
+        max_chord_surf = []
+        min_chord_surf = []
+        for zeta in self.tsaero0.zeta:
+            zeta_proj = np.zeros_like(zeta)
+            _, m, n = zeta.shape
+            for i_m in range(m):
+                for i_n in range(n):
+                    zeta_proj[:, i_m, i_n] = zeta[:, i_m, i_n].dot(self.u_ext_direction) * self.u_ext_direction
+            max_chord_surf.append(np.max(zeta_proj))
+            min_chord_surf.append(np.min(zeta_proj))
+        return min(min_chord_surf), max(max_chord_surf)
+
+    def assemble(self):
```

Comment:
It seems like nothing will work if the inherited class does not define the method assemble?? If that's the case just make the class abstract and declare the method as 
from abc import ABC, abstracmethod 
LinearGust(ABC):
...
@abstractmethod
def assemble(self):

Reply:
Good point!